pkg/hubble/metrics/metrics.go

```diff
@@ -132,44 +119,52 @@ func initEndpointDeletionHandler() {
 	}()
 }
 
-// initMetrics initializes the metrics system
-func initMetrics(address string, metricsTLSConfig *certloader.WatchedServerConfig, enabled api.Map, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) (<-chan error, error) {
-	e, err := initMetricHandlers(enabled)
+// InitMetrics initializes the metrics system
+func InitMetrics(reg *prometheus.Registry, enabled api.Map, grpcMetrics *grpc_prometheus.ServerMetrics) error {
+	e, err := initMetricHandlers(reg, enabled)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	enabledMetrics = e
 
-	registry.MustRegister(grpcMetrics)
-	registry.MustRegister(LostEvents)
-	registry.MustRegister(RequestsTotal)
-	registry.MustRegister(RequestDuration)
+	reg.MustRegister(grpcMetrics)
+	reg.MustRegister(LostEvents)
+	reg.MustRegister(RequestsTotal)
+	reg.MustRegister(RequestDuration)
 
-	errChan := make(chan error, 1)
-
-	initMetricsServer(address, metricsTLSConfig, enableOpenMetrics, errChan)
 	initEndpointDeletionHandler()
 
-	return errChan, nil
+	return nil
 }
 
-// EnableMetrics starts the metrics server with a given list of metrics. This is the
-// function Cilium uses to configure Hubble metrics in embedded mode.
-func EnableMetrics(log logrus.FieldLogger, metricsServer string, metricsTLSConfig *certloader.WatchedServerConfig, m []string, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) error {
-	errChan, err := initMetrics(metricsServer, metricsTLSConfig, api.ParseMetricList(m), grpcMetrics, enableOpenMetrics)
+func EnableMetrics(log logrus.FieldLogger, metricsTLSConfig *certloader.WatchedServerConfig, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) {
+	srv := http.Server{
+		Addr:    option.Config.HubbleMetricsServer,
+		Handler: nil,
+	}
+	errChan := make(chan error, 1)
+
+	InitMetricsServerHandler(&srv, Registry, option.Config.EnableHubbleOpenMetrics)
+	err := InitMetrics(Registry, api.ParseMetricList(option.Config.HubbleMetrics), grpcMetrics)
 	if err != nil {
-		return fmt.Errorf("unable to setup metrics: %w", err)
+		log.WithError(err).Error("Unable to setup metrics: %w", err)
+		return
 	}
+
+	go func() {
+		if metricsTLSConfig != nil {
+			srv.TLSConfig = metricsTLSConfig.ServerConfig(&tls.Config{ //nolint:gosec
+				MinVersion: serveroption.MinTLSVersion,
+			})
+			errChan <- srv.ListenAndServeTLS("", "")
+		} else {
+			errChan <- srv.ListenAndServe()
+		}
+	}()
 	go func() {
 		err := <-errChan
 		if err != nil && !errors.Is(err, http.ErrServerClosed) {
-			log.WithError(err).Error("Unable to initialize Hubble metrics server")
+			log.WithError(err).Error("Unable to start Hubble metrics server")
 		}
 	}()
-	return nil
```

Comment:
I just noticed this no longer returns the error. This is a behavior change I think we need to consider. This will mean that if metrics setup fails, then the rest of Hubble continues to start. We've actually discussed doing this previously, but I'm not sure if this is what we want. @cilium/sig-hubble any thoughts?